enhancements/network-observability.md

```diff
@@ -0,0 +1,367 @@
+to find out other information about the user, including possibly the user's
+identity.
+
+3. Cost<br>
+    It may not be clear how much additional cost the customer will incur,
+particularly if this is managed by Red Hat.  Additional nodes may need to
+be added to avoid the resource risk described above.
+
+4. Data accuracy<br>
+    If data sampling is turned on, then data accuracy cannot be relied on
+so using this for audit reports or criminal investigation are not viable.
+
+
+## Design Details
+
+### Software Architecture
+This is the software architecture at a high level.
+
+![architectural diagram](arch22.drawio.svg)
+
+The components with hatch marks indicate work that needs to be done.
+
+For the first release, it intentionally leaves out the message broker,
+the plan being Kafka, for simplicity.  For the specific choice of
+component, separate enhancements will provide the detail analysis on
+these decision.
+
+
+### Software Components
+
+#### OVS
+Open vSwitch (OVS) is the NetFlow exporter in the OpenShift cluster.  When
+Network Observability is enabled, each OVS in the node will be configured to
+send IPFIX data.  This is an all-or-none proposition, meaning there will
+not be an option to selectively choose some OVS to be exporters only.  There
+will be a configuration parameter to set the sampling rate.  Other parameters
+such as the cache size may also be exposed.
+
+#### Flow Collector
+The Flow Collector is responsible for receiving and normalizing the data
+sent by the NetFlow exporter.  This component is [GoFlow 2](https://github.com/netsampler/goflow2).
+The enhancement is at [Enable flows collection](https://github.com/mariomac/enhancements/blob/enable-flows-collection/enhancements/network/enable-flows-collection.md).
+
+#### Flow Enricher
+The Flow Enricher enhances the IPFIX data with Kubernetes-related
+information.  It watches for updates on pods, services and daemonsets
+using the Kubernetes API.  It merges data from Flow Collector and
+Kubernetes to be able to display pod-to-pod traffic.  It is also
+responsible for writing out the data to Storage.  See Network Observability
+Flow Enricher enhancement for more details.
+
+#### Storage
+Storage stores the IPFIX/Kubernetes data.  It provides a REST API to
+retrieve this data.  The minimum storage requirement is 200 GB.  Because
+of the potential large amount of data, the data retention policy is 24
+hours.  This component is [Grafana Loki](https://github.com/grafana/loki).
+See Network Observability Storage enhancement for more details.
+
+
+### Operators
+Two new operators will be required for network observability.  They are
+the Network Observability Operator and the Loki Operator.
+
+#### Network Observability Operator (NetObserv)
+The Network Observability Operator will need to be installed from OperatorHub
+to enable this feature.  This operator has a dependency on the Loki
+Operator.  This is an OpenShift Console dynamic plugin that is responsible
```

Comment:
What is a console dynamic plugin?  Not the operators ... so not sure what this refers to exactly.

Do you mean to say that the operator includes a console dynamic plugin?  I'd look into how dynamic plugins are delivered, as I think they may need to be included in the console and not delivered as part of the operator?  I'm not positive on that though.

It would help for this section to more clearly explain what the new operator will do to justify it.  So far, I can guess:

- ensure the loki operator is installed?
- ensure a loki instance is installed and configured properly?
- automatically turn on and configure flow export from OVS?